### DISCUSSION TITLE
iMXRT1024 evk

### DISCUSSION THREAD

**Discussion by phil-barrett** [terjeio/grblHAL#132] [General]
@moffy  I see this new [EVK available](https://www.nxp.com/design/development-boards/i-mx-evaluation-and-development-boards/i-mx-rt1024-evaluation-kit:MIMXRT1024-EVK).  Still a little pricey($70) but moving in the right direction. However, searching for the 1024 processor only gives me the EVK hits. I confess that NXP processor marketing confuses me. How does the 1024 differ from the 1021?  I assume they are both in the 1020 family.

**Comment by Moffy** [comment]
Yes I noticed it also, from NXP they say:
"The i.MX RT1024 expands the i.MX RT series by providing a small, low-cost LQFP package, further simplifying board design and layout for customers. The i.MX RT1024 MCU runs on the Arm® Cortex®-M7 core up to 500 MHz, provides 4 MB on-chip flash and 256 KB on-chip SRAM."

So it has on chip flash, 4Mbytes, that's cool. Looks like that's the major difference but it would free up some pins, maybe 7-8, for general I/O and simplify the board a little, no QSPI Flash required. Definitely the one to choose.

**Comment by Moffy** [comment]
Phil if you are interested, I am discussing writing the RT1021 driver for grblHAL with terjeio, as well as including autolevelling at: https://github.com/Moffy/grblHAL/issues/1. FYI, your input would be appreciated. Cheers.
